### ISSUE TITLE
Inconsistent behavior when exporting config secret

### ISSUE THREAD

**Issue by mastoj** [pulumi/pulumi#8503]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Secrets in config seems to be stored unencrypted when read from config and then directly exported as output.

1. Create new typescript project
2. Update index.ts to

```
import * as pulumi from "@pulumi/pulumi";

const config = new pulumi.Config();
export const shouldBeSecret = config.get("msg");
```

3. Add config secret: `pulumi config set msg yolo --secret`
4. Run `pulumi up`

Expected: I would expect the output to be stored encrypted.
Actual: It looks like the output is stored as a secret, but when reading the output it is in clear text (as you show in the picture):

![image](https://user-images.githubusercontent.com/501424/143723404-dd8b5148-c7b1-4497-aae5-bdb8b6420e27.png)

**Comment by stack72** [comment]
Hi @mastoj 

There's actually a known issue here where we were going to warn when someone used get for a secret

If you change the func to be getSecret then you will get the correct behaviour.

I'll leave this open so we can track fixing up that get functionality

Hope this helps

Paul

**Comment by mastoj** [comment]
Yeah, I noticed that getSecret works… but as a consumer of the sdk I think the behavior should be independent of it is a secret or not if possible.
